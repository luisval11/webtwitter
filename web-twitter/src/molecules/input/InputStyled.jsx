import React from "react";
import PropTypes from "prop-types";
import { Form } from "react-bootstrap";
import { FormLabelStyled } from "../../atoms/FormLabelStyled";

export const InputStyled = ({
	index,
	className,
	handleInput,
	label,
	name,
	...rest
}) => {
	return (
		<Form.Group controlId={index} className={className}>
			{label && <FormLabelStyled htmlFor={name} text={label} />}
			<Form.Control
				id={name}
				onChange={(event) => {
					handleInput && handleInput(event);
				}}
				{...rest}
			/>
		</Form.Group>
	);
};

InputStyled.propTypes = {
	index: PropTypes.any,
	className: PropTypes.string,
	handleInput: PropTypes.func,
	label: PropTypes.string,
	name: PropTypes.string,
	rest: PropTypes.object,
};

export default InputStyled;
