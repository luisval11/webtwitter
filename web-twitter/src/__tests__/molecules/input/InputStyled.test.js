import { mount } from "enzyme";
import { InputStyled } from "../../../molecules/input/InputStyled";

describe("Test on InputStyled", () => {
	test("should match initial snapshot", () => {
		let wrapper = mount(
			<InputStyled
				as="textarea"
				className="input__textarea"
				handleInput={() => console.log("Dummy function")}
				name="content"
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should render an input of type 'textarea'", () => {
		const typeParam = "textarea";
		let wrapper = mount(<InputStyled as={typeParam} />);
		const lenght = wrapper.find("textarea").length;

		expect(lenght).toBe(1);
	});
});

describe("Test on InputStyled with Label", () => {
	const label = "A single Label";

	test("should match initial snapshot", () => {
		let wrapper = mount(
			<InputStyled
				as="textarea"
				label={label}
				handleInput={() => console.log("Dummy function")}
				name="content"
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should render an input of type 'textarea' and a label", () => {
		const typeParam = "textarea";
		let wrapper = mount(<InputStyled as={typeParam} label={label} />);
		const lenghtTextarea = wrapper.find("textarea").length;
		const lenghtLabel = wrapper.find("label").length;

		expect(lenghtTextarea).toBe(1);
		expect(lenghtLabel).toBe(1);
	});
});
