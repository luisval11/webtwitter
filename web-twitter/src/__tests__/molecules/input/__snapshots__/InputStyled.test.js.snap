// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on InputStyled should match initial snapshot 1`] = `
<div
  className="input__textarea"
>
  <textarea
    className="form-control"
    id="content"
    onChange={[Function]}
  />
</div>
`;

exports[`Test on InputStyled with Label should match initial snapshot 1`] = `
<div>
  <label
    className="form-label"
    htmlFor="content"
  >
    A single Label
  </label>
  <textarea
    className="form-control"
    id="content"
    onChange={[Function]}
  />
</div>
`;
